src/protocols/DRMLease.cpp

```diff
@@ -246,14 +246,12 @@ CDRMLeaseProtocol::CDRMLeaseProtocol(const wl_interface* iface, const int& ver,
         if (primaryDevice->success)
             break;
     }
-
-    if (!primaryDevice || primaryDevice->success) {
-        PROTO::lease.reset();
-        return;
-    }
 }
 
 void CDRMLeaseProtocol::bindManager(wl_client* client, void* data, uint32_t ver, uint32_t id) {
+    if (!success())
```

Comment:
this is wrong. We cannot refuse binding because that will crash clients.

We _have to_ remove the global. Which, IMO, should be done by destroying this protocol object

Reply:
yeah but isnt that what we do in ProtocolManager.cpp line 159. if the fd fails to dup we both dont bind and we destroy the protocol object. if it succeeds it binds and doesnt destroy

Reply:
destroy it in doLater then

Reply:
wont that cause a possible race with us binding the protocol on a failed fd? since doLater is technically inside the wayland eventloop? and some client trying to use this while it cant, just "thinking" out load heh

Reply:
roger, il change it back to dolater etc. give me a few minutos

Reply:
now, we are down to very few line changes heh